Sources/Event+Identity.swift

```diff
@@ -0,0 +1,27 @@
+//
+// Copyright 2022 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may obtain a copy
+// of the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under
+// the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+// OF ANY KIND, either express or implied. See the License for the specific language
+// governing permissions and limitations under the License.
+//
+
+import AEPCore
+import Foundation
+
+extension Event {
+
+    /// Reads the advertising from the event data
+    var adId: String? {
```

Comment:
What do you think about keeping it as optional where the `nil` value case is returned if the event fails the isAdIdEvent check? This would allow us to distinguish between `adId` values where it was valid, and it was accessed even though it is not an AdId event. There are some cases where `adId` is used without first checking `isAdIdEvent`. If this is publicly accessible as well, I think having the check automatically built in will prevent a lot of erroneous usage.

Reply:
I think it is not necessary to do this check twice if we call updateAdvertisingIdentifier only when the event is valid, this way we can keep the two method independent and concise.

Reply:
I totally agree, in the current flow it is definitely double work. Do you know if this Event+Identity extension available to users of the public APIs as well? If not then I'll update the `adId` to just do the extraction logic without depending on `isAdIdEvent`

Reply:
The access level has been updated back to private, and data type is non-optional String